Library/Homebrew/cask/cask.rb

```diff
@@ -131,18 +139,40 @@ def caskfile_only?
 
     sig { returns(T.nilable(Time)) }
     def install_time
-      _, time = timestamped_versions.last
-      return unless time
-
-      Time.strptime(time, Metadata::TIMESTAMP_FORMAT)
+      time = installed_caskfile&.dirname&.dirname&.basename&.to_s
```

Comment:
A comment showing what part of the path you're going for here could be useful. Basically like what you have in `installed_version`.